### ISSUE TITLE
reading temperature / workaround for defective ICs

### ISSUE THREAD

**Issue by ghost** [adafruit/Adafruit_CircuitPython_DPS310#5]
[issue mistakenly posted to adafruit / Adafruit_DPS310 - reposting here]

First, thanks for this library!

I have a DPS310 on a grove module, it's not from Adafruit so maybe my issue doesn't apply to ICs shipped by Adafruit. Anyway - as mentioned in Infineon's [official lib ](https://github.com/Infineon/DPS310-Pressure-Sensor#known-issues) some ICs have a problem with temperature (mine does) in which case their "correctTemp()" function has to be called just after i2c initilaization (or as tests showed, after a reset).

The following fix worked for me. I'm a noob with python so I didn't want to modify the lib and send a pull request - instead I've just overriden the _reset() function. Maybe add that fix to the library ?

~~~
class fixDPS310(DPS310):

    # magic values from the official library's DpsClass.cpp
    _reg0e = RWBits(8, 0x0E, 0)
    _reg0f = RWBits(8, 0x0F, 0)
    _reg62 = RWBits(8, 0x62, 0)

    def _correct_temp(self):
        self._reg0e = 0xA5
        self._reg0f = 0x96
        self._reg62 = 0x02
        self._reg0e = 0
        self._reg0f = 0
        unused_var = self._raw_temperature

    def _reset(self):
        # print('override reset')
        self._reset_register = 0x89
        # wait for hardware reset to finish
        sleep(0.010)
        while not self._sensor_ready:
            sleep(0.001)
        self._correct_temp()

dps310 = fixDPS310(i2c)
# [...]
~~~

**Comment by ladyada** [comment]
hi, thanks for the fix! a PR would be most welcome and we can help get it merged - especially since it seems not all DPS's have this issue :) would you be interested in that?

**Comment by ghost** [comment]
Sure, I can try to send a PR ; the reason for not doing so in the first place is because I'm a noob with python so I wasn't sure my "coding style" was OK, nor the way I used the RWBits() function. I'll send it over the next few days.
By the way I've also noticed that the IC had to be reset with _reset() before being able to change its mode with mode(). Do you prefer that I send a PR fixing both the temperature reading and (re)setting the mode, or do you prefer two separate (incremental) PRs ?

**Comment by ladyada** [comment]
both is fine!

**Comment by evaherrada** [comment]
Fixed by #6. Closing
